File: backend/src/main/java/com/serenade/backend/domain/playlist/dto/PlaylistDetailResponse.java
package com.serenade.backend.domain.playlist.dto;

import com.serenade.backend.domain.track.dto.TrackResponse;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record PlaylistDetailResponse(
        UUID id,
        String name,
        boolean isCopy,
        UUID sourcePlaylistId,
        int version,
        double ratingAvg,
        List<TrackResponse> tracks,
        Instant updatedAt
) {}

File: app/app/src/main/java/com/serenade/app/feature/playlist/data/remote/dto/PlaylistDetailResponse.kt
package com.serenade.app.feature.playlist.data.remote.dto

import com.serenade.app.feature.track.data.remote.dto.TrackResponse
import kotlinx.serialization.Serializable

@Serializable
data class PlaylistDetailResponse(
    val id: String,
    val name: String,
    val isCopy: Boolean,
    val sourcePlaylistId: String? = null,
    val version: Int,
    val ratingAvg: Double,
    val tracks: List<TrackResponse>,
)
